### PR TITLE
[INFINITY-1877] Add Secrets Tutorial

### PR DESCRIPTION
**Pull request by mbdcos**
I have been preparing a tutorial with a real application such as NGINX secure web server. There is a branch ready about secure-NGINX with secrets.
The problem is that we need to create a new framework for this tutorial. And we do not have a framework generation procedure. Generating a new framework from template is not automated and it requires a lot of explanation.
Eventually, secret tutorial would become more about how to write a standalone framework using SDK.
Since we do not have a automated and easy way to create a new framework, and we do not want to go into all build and universe details about how to generate a framework, it is best to skip secure-NGINX example, and use something that already exists in dcos-commons, such as hello-world for secrets tutorial.

### CODE REVIEW

**Comment by mbdcos** [comment]
Instead of a sample RSA key/cert we can also do as follows:

https://github.com/mesosphere/dcos-commons/blob/3b6bdd2f81a43dac49e6c01f1b5f4265579da41d/docs/pages/tutorials/secrets-tutorial.md

**Comment by mbdcos** [comment]
ping @ArtRand @keithchambers @sascala

File: docs/pages/tutorials/secrets-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+Now, lets create a key and add its private key to a secret with path `hello-world/secret2` and its public key to another secret with path `hello-world/secret3`.
+
+Create key and cert:
+
+```
+$openssl genrsa -out privatekey.pem 1024
+$openssl req -new -x509 -key privatekey.pem -out publickey.cer -days 1825
+```
+
+Create secrets with value from these private and public key files:
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: Create secrets with the values from the private and public...

File: docs/pages/tutorials/secrets-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+```
+$ dcos security secrets list  hello-world
+```
+```
+$ dcos security secrets get hello-world/secret3
+```
+
+
+## Install Service with Secrets
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: Install the Service

File: docs/pages/tutorials/secrets-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+
+
+The `hello-world` package includes a sample YAML file for secrets. Please examine the following `examples/secrets.yml` file to grasp how secrets are declared:
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: The following file demonstrates how secrets are declared:

File: docs/pages/tutorials/secrets-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        goal: RUNNING
+        cmd: >
+               env &&
+               ls -la &&
+               cat WORLD_SECRET*_FILE &&
+        ................
+```
+
+
+The `hello` pod has two secrets. The first secret with path `hello-world/secret1` is exposed both as an environment variable and as a file. The second one is exposed only as a file. The value of the second secret with path `hello-world/secret2`, that is the private key, will be copied to the `HELLO_SECRET2_FILE` file located in the sandbox.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: The first secret, with path `hello-world/secret1`,

, which is the private key,

File: docs/pages/tutorials/secrets-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+               env &&
+               ls -la &&
+               cat WORLD_SECRET*_FILE &&
+        ................
+```
+
+
+The `hello` pod has two secrets. The first secret with path `hello-world/secret1` is exposed both as an environment variable and as a file. The second one is exposed only as a file. The value of the second secret with path `hello-world/secret2`, that is the private key, will be copied to the `HELLO_SECRET2_FILE` file located in the sandbox. 
+  
+The `world` pod has three secrets. The first one is exposed only as an environment variable. The second and third secrets are exposed only as files. All `server` tasks in the `world` pod will have access to the value of these three secrets, either as a file and/or as an evironment variable.  Please note that the secret path is the default file path if no `file` keyword is given. Therefore, the file path for the third secret is same as the secret path.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: or as an environment variable

**Note:** The secret path is the default...

File: docs/pages/tutorials/secrets-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+
+Next, install `hello-world` package using the following "`option.json`" file:
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: install the
`option.json`

File: docs/pages/tutorials/secrets-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+      },
+      "world":{
+          "secret1": "hello-world/secret1",
+          "secret2": "hello-world/secret2",
+          "secret3": "hello-world/secret3"
+          }
+}
+```
+
+Here, we use `examples/secrets.yml` spec file. And, we also overwrite a couple of `hello` and `world` options -  `secret1`, `secret2`, and `secret3` parameters are set to specific secret paths.  Please examine `universe/config.json` for more information about the `hello-world` package configuration.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: We also overwrite several `hello`...

Examine `universe/config.json`....

File: docs/pages/tutorials/secrets-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+
+
+## Verify Secrets
+
+
+You can use `dcos task exec` command to attach to the container that is running the task you want to examine.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: Use the `dcos task exec`...

File: docs/pages/tutorials/secrets-tutorial.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+Here, we use `examples/secrets.yml` spec file. And, we also overwrite a couple of `hello` and `world` options -  `secret1`, `secret2`, and `secret3` parameters are set to specific secret paths.  Please examine `universe/config.json` for more information about the `hello-world` package configuration.
+
+
+
+## Verify Secrets
+
+
+You can use `dcos task exec` command to attach to the container that is running the task you want to examine. 
+
+Please note that tasks of the `hello` pod in our example are running inside a docker image (`ubuntu:14.04`).
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: **Note:** The tasks of the ...

File: docs/pages/tutorials/secrets-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+> $ cat hello-world/secret3 
+..............................................
+..............................................
+  <CERT value created via openssl>
+..............................................
+..............................................
+```
+
+
+Please note that the third secret for the `word` pod does not have a specific `file` keyword. Therefore, its secret path `hello-world/secret3` is also used as the file path by default. As can be seen in the output, `hello-world` directory is created and content of the third secret is copied to the `secret3` file.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: **Note:** The third secret...

..the content of the third secret

File: docs/pages/tutorials/secrets-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+
+We can update the content of a secret as follows:
+
+```
+$ dcos security secrets update -v "This is the NEW value for secret1" hello-world/secret1
+```
+
+
+
+Secret value is securely copied from the secret store to a file and/or to an environment variable. A secret file is an in-memory file and it disappears when all tasks of the pod terminate.
```
Comment (sascala) [inline docs/pages/tutorials/secrets-tutorial.md]: The secret value is...

**Pull request merged**